### PR TITLE
Assertion error Fix - Fixed Assertion true === false error

### PR DESCRIPTION
**Pull request by deveshmehta7**

### CODE REVIEW

**Comment by JamesIves** [comment]
[This bug](https://github.com/spdy-http2/spdy-transport/issues/47) has been causing some problems for our team. I went ahead and tested the changes in this PR locally and they seem to be resolved. Would love to see this get properly reviewed/merged by the maintainers.

**Comment by Ailuue** [comment]
Our team also uses an HTTPS app locally, and constantly crashes due to the error this fixes. Please review/merge ASAP!

**Comment by Miz85** [comment]
+1 It'd be really great to get this fix merged. Many thanks to all people who are involved in this btw!

File: lib/spdy-transport/priority.js [hunk 1]
```diff
@@ -71,10 +71,13 @@ PriorityNode.prototype.remove = function remove () {
 PriorityNode.prototype.removeChild = function removeChild (child) {
   this.children.weight -= child.weight
   var index = utils.binarySearch(this.children.list, child, compareChildren)
-  assert(index !== -1)
-
-  // Remove the child
-  this.children.list.splice(index, 1)
+  try {
```
Comment (strange-developer) [inline lib/spdy-transport/priority.js]: Not an author or reviewer but I'm stuck with this issue as well. I just noticed this logic could be done a bit easier and without the `try catch`:

```js
if (index !== -1 && this.children.list.length >= index) {
  this.children.list.splice(index, 1);
}
```

**Comment by d-mg** [comment]
Would be nice if someone could merge this in.

**Comment by nikvaessen** [comment]
+1

**Comment by YoniH** [comment]
Anyone merging this soon?
I get this error from Create React App and I don't want to eject my project just to fiddle with webpack and work around this

**Comment by Grantlyk** [comment]
This needs to be merged -- is anyone maintaining this repo?

**Comment by diasdavid** [comment]
Thank you @deveshmehta7 for the PR. Handling the review, merge and release now

**Comment by diasdavid** [comment]
Published in spdy-transport in 2.1.1. Apologies to everyone on the delay on getting to this one, as described on https://github.com/spdy-http2/node-spdy/issues/339, I'm beyond capacity to be capable to maintain this module, I hope you understand.

**Comment by niftylettuce** [comment]
thank you @diasdavid

**Comment by niftylettuce** [comment]
@diasdavid 	can we get this version bumped too https://github.com/spdy-http2/node-spdy/blob/master/package.json#L42?

**Comment by diasdavid** [comment]
@niftylettuce semver should pick on it, right?

**Comment by mgol** [comment]
@niftylettuce Remove `node_modules` and `package-lock.json` and run `npm install` again and you should get it.

**Comment by niftylettuce** [comment]
ty

**Comment by raRaRa** [comment]
Thank you so much for the fix. This has been annoying me for the past weeks.

A small addition to @mgol answer: For people using `yarn`,  you need to delete `yarn.lock` instead of `package-lock.json`.

**Pull request merged**